### PR TITLE
Merge upstream changes

### PR DESCRIPTION
**Pull request by dyroffk**
- CustomFields support was added
- Authorization header changes which replace https://github.com/Vidyard/ruby-pardot/pull/10
- Escaping URL entities
- Updated gem dependencies

Merging in upstream changes before adding Salesforce SSO authentication support

### CODE REVIEW

File: lib/pardot/http.rb [hunk 1]
```diff
@@ -5,8 +5,9 @@ def get object, path, params = {}, num_retries = 0
       headers = {}
       smooth_params_and_headers object, params, headers
       full_path = fullpath object, path
+      headers = create_auth_header object
       check_response self.class.get(full_path, :query => params, :headers => headers)
-
+
```
Comment (avonhatten1996) [inline lib/pardot/http.rb]: Might want to get rid of these whitespaces

File: spec/support/fakeweb.rb [hunk 2]
```diff
@@ -12,3 +12,7 @@ def fake_get path, response
 def fake_authenticate client, api_key
   client.api_key = api_key
 end
+
+def assert_authorization_header
+  expect(FakeWeb.last_request[:authorization]).to eq('Pardot api_key=my_api_key, user_key=bar')
+end
```
Comment (avonhatten1996) [inline spec/support/fakeweb.rb]: Missing the EOF here

**Review by avonhatten1996 — CHANGES_REQUESTED** [review]
Couple formatting nits then good to go

**Comment by dyroffk** [comment]
ok @avonhatten1996 I addressed the comments.  I think I will follow up and add linting to this repo since there were a number of other violations when I ran it locally

I'm going to test this in staging to sanity check the merge went OK

**Review by avonhatten1996 — APPROVED** [review]

**Review by karshell — APPROVED** [review]

**Review by bdmubara — APPROVED** [review]

**Comment by dyroffk** [comment]
Standby looks like I need to address some differences between `smooth_params_and_headers` and `smooth_params` in this repo vs the upstream.  Ran into a few errors while testing.  Marking blocked until I finish addressing and it's ready for re-review

**Comment by dyroffk** [comment]
OK this is ready for review again.  I tested end to end in sandbox environment and confirmed working as expected

**Comment by dyroffk** [comment]
2021-01-25-144025_2131x538_scrot

Fixes included in this changeset address email addresses with special characters failing to create prospects in Pardot

**Pull request merged**